hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsScheduler.java

```diff
@@ -127,6 +126,10 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
       status.getMessage(),
       liveState.getStagingTasksSize()));
 
```

Comment:
@elingg: This pattern of having the Observervable determine what information the Observers should receive is a bad idea.  This is emblematic of one of the pervasive problems in the over all implementation of HdfsScheduler.  It is constantly inspecting and interacting with the internal state of other Classes, in particular LiveState and PersistentState.  I'm trying to get away from this mistake with the Observer pattern.

The motivation for limiting the information provided to the Observers is also committing the sin of premature optimization.  We're talking about tens of events here and are worried that handling them (in memory) is going to slow things down.  It's a non-issue and shouldn't be a motivation for inspecting the internal state of Observers.

In my view wer're degrading cohesion and breaking any sense of layering for no clear win.